CHANGELOG.md

```diff
@@ -14,6 +14,17 @@
     - udf
 - Added the function `DataFrame.write.csv` to unload data from a ``DataFrame`` into one or more CSV files in a stage.
 
+### Behavior change
+
+- Changed behavior when reading file from stage. When reading a path that does not exist or when there are no files under the path, `FileNotFound` is raised instead of `SnowflakeSQLException`. This behavior change applies to APIs:
+  - snowflake.snowpark.DataFrameReader:
```

Comment:
nit: `FileNotFoundError`.

Also you should mention that csv also displays this behavior, we are aligning the other api with csv and fixing the additional error raised by csv for empty files.